src/java/org/apache/cassandra/index/sai/README.md

```diff
@@ -0,0 +1,116 @@
+<!---
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-->
+
+# Storage-Attached Indexing
+
+## Overview
+Storage-attached indexing is a column based secondary indexing apparatus for Cassandra.
+
+The project was inspired by SASI (SSTable-Attached Secondary Indexes) and retains some of its high-level
+architectural character (and even some actual code), but makes significant improvements in a number of areas:
+
+- The on-disk/SSTable index formats for both string and numeric data have been completely replaced. Strings are indexed
+  on disk using our proprietary on-disk byte-ordered trie data structure, while numeric types are indexed using Lucene's
+  balanced kd-tree.
+- While indexes continue to be managed at the column level from the user's perspective, the storage design at the column
+  index level is row-based, with related offset and token information stored only once at the SSTable level. This
+  drastically reduces our on-disk footprint when several columns are indexed on the same table.
+- The query path is synchronous and index searches run on IO threads.
```

Comment:
The first part of this makes sense, but I can't remember what we meant by "index searches run on IO threads" 🤔

Reply:
I'm going to remove that section. It looks to be another throwback to TPC and makes no sense with the current architecture.